From: Michael Tokarev <mjt@tls.msk.ru>
Date: Thu, 19 May 2022 20:37:21 +0300
Subject: add missing libs deps
Bug-Debian: https://bugs.debian.org/1010922

Lots of samba libraries has incomplete dependencies listed
in wscript files.  This usually is not a problem since the
link line includes dependencies of their dependencies of
their dependencies, and somewhere down that line all immediate
dependencies which are missing are actually present.  But
sometimes this becomes a problem when a library does not
declare direct dependency on at least one private library
which it actually uses: in case no private library is
listed as direct dependency, private library directory is
not put into RUNPATH of the resulting binary, so the binary
can not find its own dependencies.

Fix a few such places, including one library which is a part
of public abi (libsmbldap).

diff --git a/lib/util/wscript_build b/lib/util/wscript_build
index 2f31e8fa5b1..08a77b8940c 100644
--- a/lib/util/wscript_build
+++ b/lib/util/wscript_build
@@ -218,5 +218,5 @@ else:
     bld.SAMBA_LIBRARY('samba-modules',
                       source='modules.c',
-                      deps='samba-errors samba-util',
+                      deps='samba-errors samba-util samba-debug',
                       local_include=False,
                       private_library=True)
diff --git a/source3/wscript_build b/source3/wscript_build
index acfc0c56f03..e919f38d9cb 100644
--- a/source3/wscript_build
+++ b/source3/wscript_build
@@ -189,5 +189,5 @@ bld.SAMBA3_LIBRARY('smbldaphelper',
                           passdb/pdb_ldap_util.c
                           ''',
-                   deps='smbldap secrets3',
+                   deps='smbldap secrets3 replace',
                    allow_undefined_symbols=True,
                    enabled=bld.CONFIG_SET('HAVE_LDAP'),
@@ -483,5 +483,5 @@ bld.SAMBA3_LIBRARY('secrets3',
 bld.SAMBA3_LIBRARY('smbldap',
                     source='lib/smbldap.c',
-                    deps='ldap lber samba-util smbconf',
+                    deps='ldap lber samba-util smbconf replace samba-debug samba-security',
                     enabled=bld.CONFIG_SET("HAVE_LDAP"),
                     private_library=False,
